src/test/runtests.sh

```diff
@@ -4,8 +4,29 @@
 
 TotalErrors=0
 
+
+# Syslog is required for some ut_outmdsd tests.
+# In docker image, rsyslogd is not started and may not
+# configured properly. So handle them in this function.
+#
+# In non-docker machine, assume the proper syslog is
+# configured. If not, user may get test error, and test
+# error should contain details on what's wrong.
+ConfigSyslog()
+{
+    grep docker /proc/self/cgroup > /dev/null
+    if [ $? == 0 ]; then
+        echo Configure syslog ...
+        sudo chown syslog.syslog /var/log/
+        sudo rsyslogd > /dev/null 2>&1
+        sudo chmod a+r /var/log/syslog
+    fi
```

Comment:
This is a race. If the script tries to do the chmod before rsyslogd actually gets around to creating the file, this script will fail.
Before you start rsyslogd, run "sudo touch /var/log/syslog" to ensure the file exists. rsyslogd will append to it.